Add formatWithSplit class of functions to resolve issue #7.

4onen opened the pull request:
Let me know if all of the convenience functions this adds clutter up the documentation -- the core of this change is the `formatSplitWith` function and corresponding rewrite of formatWith.

4onen committed:
Add formatWithSplit class of functions, to resolve issue #7.

domenkozar commented:
Nice! @basti1302 what do you think?

basti1302 commented:
Sorry, this has been sitting for a while. Good stuff! Could you add a few tests for the new functions? Then I'll merge it and publish it.

4onen commented:
Sorry for the delay in getting back to this; I was out travelling.

Unfortunately, I (and a number of other MacOS users) have had less than stellar luck getting elm-test to run since the update to Elm 0.19. I had some luck with elm-test 0.19 beta2, but since then the install scripts have consistently failed on my machine.

I am familiar with elm-test however, and could write the tests, although I'd be unable to verify their functionality apart from type-correctness with the elm compiler. Is that acceptable, @basti1302?

basti1302 commented:
I'm also a MacOS and must say I had a flawless experience when migrating the tests of https://github.com/basti1302/elm-non-empty-array to 0.19 - with version 0.19.0-rev3 of the node test runner. That version came out only a few days ago, so if you have issues with the previous versions, it might be worth checking out that latest version at some point.

That said, for this PR your suggested strategy would also be perfectly okay. So if you could add a few tests that at least compile, I'm willing to take it from there. Besides, Travis would also check them anyway ;-)

4onen committed:
Write tuple-based tests, correct tuple-based Filesize exports.

4onen commented:
All done! Again, sorry for the delay; Thanksgiving celebrations happened. On the plus side, I managed to get elm-test rev3 installed! Turns out my npm install had some weird permissions errors. Ugh.

Anyway, the tests are a copy of the string-based tests, just converted to tuples. That should cover all of the same cases without being too confusing. In getting those tests built, I also discovered a serious (now-corrected) mistake in my PR: I never exported the new functions! D'oh! That is also corrected by this commit.

Let me know if I fumbled anything else, but for now Travis agrees with my local runner; seems good now!

basti1302 commented:
Looks splendid! Absolutely no need to be sorry about any delays, after all I ignored this for much longer. I had to add the new functions also to the module documentation (the `@docs` statement), otherwise `elm-diff`/`elm-publish` would complain, but except for this really minor omission, this can go out as is.

Landed in release 1.2.0 .
